src/datasets/arrow_dataset.py

```diff
@@ -780,7 +771,7 @@ def from_pandas(
             info = DatasetInfo()
         info.features = features
         table = InMemoryTable.from_pandas(
-            df=df, preserve_index=preserve_index, schema=pa.schema(features.type) if features is not None else None
+            df=df, preserve_index=preserve_index, schema=features.arrow_schema if features is not None else None
```

Comment:
This is the new way of getting the arrow schema of some features.
Contrary to the old way, it also incorporates the HF metadata about the actual feature types the schema represeent